cmd/operator-opamp-bridge/internal/proxy/agent.go

```diff
@@ -0,0 +1,170 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package proxy
+
+import (
+	"sync"
+
+	"github.com/go-logr/logr"
+	"github.com/google/uuid"
+	"github.com/open-telemetry/opamp-go/protobufs"
+	"github.com/open-telemetry/opamp-go/server/types"
+	"google.golang.org/protobuf/proto"
+)
+
+type Agent struct {
+	// logger for the agentImpl
+	logger logr.Logger
+	// Agent's instance id. This is an immutable field.
+	InstanceId uuid.UUID
+	// Connection to the Agent.
+	conn types.Connection
+	// mutex for the fields that follow it.
+	mux sync.RWMutex
+
+	health          *protobufs.ComponentHealth
+	effectiveConfig *protobufs.EffectiveConfig
+	// Agent's current status.
+	Status *protobufs.AgentToServer
+}
+
+func NewAgent(logger logr.Logger, agentId uuid.UUID, conn types.Connection) *Agent {
+	return &Agent{
+		logger:     logger,
+		InstanceId: agentId,
+		conn:       conn,
+	}
+}
+
+func (a *Agent) hasCapability(capability protobufs.AgentCapabilities) bool {
+	if capability < 0 {
+		return false
+	}
+	//nolint:gosec
+	return a.Status.Capabilities&uint64(capability) != 0
+}
+
+func (a *Agent) updateAgentDescription(newStatus *protobufs.AgentToServer) (agentDescrChanged bool) {
+	prevStatus := a.Status
+
+	if a.Status == nil {
+		// First time this Agent reports a status, remember it.
+		a.Status = newStatus
+		agentDescrChanged = true
+	} else {
+		// Not a new Agent. Update the Status.
+		a.Status.SequenceNum = newStatus.SequenceNum
```

Comment:
This feels like it should be outside this function. It doesn't have anything to do with the description.